src/validators/BaseValidator.ts

```diff
@@ -76,18 +77,18 @@ export abstract class BaseValidator<T> {
 	 * Sets if the validator should also run constraints or just do basic checks.
 	 * @param isValidationEnabled Whether this validator should be enabled or disabled. Set to `null` to go off of the global configuration.
 	 */
-	public setValidationEnabled(isValidationEnabled: boolean | null): this {
+	public setValidationEnabled(isValidationEnabled: boolean | (() => boolean) | null): this {
 		const clone = this.clone();
 		clone.isValidationEnabled = isValidationEnabled;
 		return clone;
 	}
 
 	public getValidationEnabled() {
-		return this.isValidationEnabled;
+		return getValue(this.isValidationEnabled);
 	}
 
 	protected get shouldRunConstraints(): boolean {
-		return this.isValidationEnabled ?? getGlobalValidationEnabled();
+		return getValue(this.isValidationEnabled) ?? getGlobalValidationEnabled();
```

Comment:
```suggestion
		return this.isValidationEnabled === null ? getGlobalValidationEnabled() : getValue(this.isValidationEnabled) ?? getGlobalValidationEnabled();
```

Reply:
if `isValidationEnabled` is null, `getValue(this.isValidationEnabled)` will return null so it'll go to `getGlobalValidationEnabled()` so do we need an extra check?

Reply:
oh right.. derpy moments